avm/res/compute/virtual-machine/main.bicep

```diff
@@ -474,35 +475,27 @@ resource vm 'Microsoft.Compute/virtualMachines@2022-11-01' = {
     storageProfile: {
       imageReference: imageReference
       osDisk: {
-        name: '${name}-disk-os-01'
-        createOption: contains(osDisk, 'createOption') ? osDisk.createOption : 'FromImage'
-        deleteOption: contains(osDisk, 'deleteOption') ? osDisk.deleteOption : 'Delete'
+        name: osDisk.name ?? '${name}-disk-os-01'
```

Comment:
I think you need to check first, if name property is set (with contains, see old code)

Reply:
Hey @rahalan - thanks for thinking with me 😃. Got it working, forgot to add the safe-dereference operator (`?`): https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/operators#safe-dereference. Some of the user-defined type properties are nullable so they need to be accessed with the safe-dereference operator else the build/deployment breaks.

I used coalesce (`??`) operator because I find it cleaner than the `contains()` function. Also, it is used this way in other Bicep AVM templates. I am OK with using contains() if you find that more suitable for this template?